pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -367,6 +367,7 @@ func getSystemImages(spec v3.ClusterSpec) (*v3.RKESystemImages, error) {
 		return nil, err
 	}
 
+	//version is set at API level during cluster create/update, but checking for empty version for backward compatability
```

Comment:
been thinking that for backwards compatibility, we should do something like:

```
version := spec.RancherKubernetesEngineConfig.Version
if version == "" {
//try getting it from the applied spec (if appliedSpec is not nil). Only if not there, fetch from settings
}
```
That would prevent upgrade for the cluster that was created w/o k8s version prior to this fix